DiagonalMatrix/DiagonalMatrix.cs

```diff
@@ -0,0 +1,98 @@
+﻿namespace DiagonalMatrixNamespace
+{
+    public class DiagonalMatrix
+    {
+        public int[] DiagonalElements { get; set; }
+        public int Size { get; }
+
+        public DiagonalMatrix(params int[] matrixElements)
+        {
+            if (matrixElements is null)
+            {
+                DiagonalElements = [];
+                Size = 0;
+            }
+            else
+            {
+                Size = matrixElements.Length;
+                DiagonalElements = new int[Size];
+                Array.Copy(matrixElements, DiagonalElements, Size);
+            }
+        }
+
+        public int this[int i, int j]
+        {
```

Comment:
Split checks for incorrect indexes and  IsElementOnMainDiagonal

Reply:
Fixed. Not calling AreIndexesCorrect(i, j) in the getter, since the task says even if the element is outside the matrix it should return 0. So calling the method here is pointless.